cpp/src/reductions/scan.cu

```diff
@@ -116,29 +120,36 @@ namespace { //anonymous
         }
 
         static
-            gdf_error exclusive_sum(void *&temp_storage, size_t &temp_storage_bytes,
-                const T *input, T *output, size_t size, cudaStream_t stream) {
-            cub::DeviceScan::ExclusiveSum(temp_storage, temp_storage_bytes,
-                input, output, size, stream);
+        gdf_error exclusive_scan(void *&temp_storage, size_t &temp_storage_bytes,
+            const T *input, T *output, size_t size, cudaStream_t stream)
+        {
+            Op op;
+            T identity = Op::template identity<T>(); // value for output[0]
+            cub::DeviceScan::ExclusiveScan(temp_storage, temp_storage_bytes,
```

Comment:
cub allocation vs RMM allocation - are we ok with this?

Reply:
I believe RMM allocation with cub is ok. @jrhemstad  @harrism any comments?

Reply:
CUB isn't allocating any memory here.

CUB requires the user to do it's temporary buffer allocation, which is being done with RMM here: https://github.com/rapidsai/cudf/blob/e68dc4353db5c774b0c3a9bc651a4f4822910306/cpp/src/reductions/scan.cu#L80